### DISCUSSION TITLE
Backend Services for Tableau websites

### DISCUSSION THREAD

**Discussion by andyl** [elixir-tools/tableau#71] [Ideas]
For a Tableau website, I'm going to write a 'Contact Service', a server-side daemon that accepts posts from an HTML form (name, email-address, comment), checks a simple 'captcha', and forwards the data to an email address.  Probably I will write the server as a Plug-based elixir script, although it could be done in Phoenix or some other language.  

It seems to me that a library of server-side utilities would be a nice complement to Tableau.  Things like contact forms, chat services, "who's online", blog comments, shopping carts, etc.

Do you agree?  

If so, what technology stack is most interesting to Tableau developers?  On the server side, there are plug-based servers, Phoenix channels, [LiveState](https://github.com/launchscout/live_state).  Probably there are commercial service providers and off-the-shelf open-source solutions.  On the client side, there is [HTMX](https://htmx.org/), [phx-live-state](https://github.com/launchscout/phx-live-state), [live-templates](https://github.com/launchscout/live-templates), raw HTML, and a bunch of JS-driven front ends.  

I'm curious if anyone else has similar interests, if there is a consensus on most-favored technology stack, and if anyone has ideas on how to organize a directory of open-source code and example solutions.

**Comment by mhanberg** [comment]
I'd probably need to see something concrete and complete with a deployment solution before having any real thoughts. 

The ultimate ergonomics would be something like Netlify functions, but sadly they don't support beam languages for there serverless functions.
